gdal/ogr/ogrsf_frmts/shape/dbfopen.c

```diff
@@ -2010,7 +2014,9 @@ DBFReorderFields( DBFHandle psDBF, int* panMap )
 
             /* load record */
             psDBF->sHooks.FSeek( psDBF->fp, nRecordOffset, 0 );
-            psDBF->sHooks.FRead( pszRecord, psDBF->nRecordLength, 1, psDBF->fp );
+            if (psDBF->sHooks.FRead(pszRecord, psDBF->nRecordLength, 1, psDBF->fp) != 1) {
+              return FALSE;
```

Comment:
memleak of several things...

Reply:
Some fairly complex state being held here, I've reworked a bit to consolidate error handling, but I'm nervous that psDBF might be left in an invalid state after a partial loop through nRecords.  Thoughts on how to handle this correctly?